exporters/geneva/src/etw_data_transport.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/exporters/geneva/metrics/etw_data_transport.h"
+#include "opentelemetry/exporters/geneva/metrics/macros.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter {
+namespace geneva {
+namespace metrics {
+#define GUID_FORMAT                                                            \
+  "%08lX-%04hX-%04hX-%02hhX%02hhX-%02hhX%02hhX%02hhX%02hhX%02hhX%02hhX"
+#define GUID_ARG(guid)                                                         \
+  (guid).Data1, (guid).Data2, (guid).Data3, (guid).Data4[0], (guid).Data4[1],  \
+      (guid).Data4[2], (guid).Data4[3], (guid).Data4[4], (guid).Data4[5],      \
+      (guid).Data4[6], (guid).Data4[7]
+
+ETWDataTransport::ETWDataTransport(const size_t offset_to_skip)
+    : offset_to_skip_{offset_to_skip} {
+  auto status =
+      ::EventRegister(&kMDMProviderGUID, NULL, NULL, &provider_handle_);
+  if (status != ERROR_SUCCESS) {
+    LOG_ERROR("ETWDataTransport:: Failed to initialize the ETW provider.  "
+              "Metrics will not be published, Provider ID: {" GUID_FORMAT "}",
+              GUID_ARG(kMDMProviderGUID));
+    provider_handle_ = INVALID_HANDLE;
+  }
+}
+
+bool ETWDataTransport::Connect() noexcept {
```

Comment:
Can EventRegister be moved to `Connect()` instead of ctor? We we could support multiple rounds of init and uninit.

Reply:
Yeah, that's a option to have once we start supporting multiple init/uninit. Will keep it for future. Thanks for the suggestion.